### ISSUE TITLE
Lines in Parallel Coordinates plot not anti-aliased on some hardware

### ISSUE THREAD

**Issue by sid-thakur** [plotly/plotly.js#2756]
The lines in Plotly's parallel coordinates plot (plotly.js) are not appearing smooth or correctly anti aliased. Is there a CSS setting or parameter to control rendering of lines in the plot? 

Below is a comparison with native D3 parallel coordinates plot. The plots have been zoomed in a bit. Thanks. 

![screen shot 2018-06-27 at 9 29 28 am](https://user-images.githubusercontent.com/5922613/41977375-27770542-79ed-11e8-8b26-29a0334a9e52.png)

**Comment by etpinard** [comment]
Hmm. https://rreusser.github.io/plotly-mock-viewer/#gl2d_parcoords doesn't look as blurry to me as your screenshot of the left. Would you mind sharing a reproducible example?

We turn on antialising [here](https://github.com/plotly/plotly.js/blob/c2457ff7d2a3189d35a190b9e4bb315f49dc080d/src/lib/prepare_regl.js#L42), so this report seems a little strange to me. What us could be going on? Maybe @dy could help us out?

> Is there a CSS setting or parameter to control rendering of lines in the plot?
> 
> Below is a comparison with native D3 parallel coordinates plot.

Our parcoord lines are drawn in WebGL not d3, so CSS won't fix the problem unfortunately.

**Comment by alexcjohnson** [comment]
@sid-thakur what hardware are you using? I have to say when I look at https://rreusser.github.io/plotly-mock-viewer/#gl2d_parcoords on my mac retina display it looks a bit blurry, as do scattergl plots like https://rreusser.github.io/plotly-mock-viewer/#gl2d_10, whereas gl3d plots look crisp. I hadn't noticed this before but I mostly display plots on my second monitor, which is not as high resolution. Perhaps we need to oversample in addition to antialiasing?

**Comment by archmoj** [comment]
As a reference point, `Mac` with `FireFox` renders like this i.e. without anti-aliasing (Note: although anti-aliasing is requested by the code on some hardware it may not be enabled). To improve this situation, such cases could be detected and doubling `glPixelRatio` values may help.
https://rreusser.github.io/plotly-mock-viewer/#gl2d_parcoords
![macOS_with_FireFox](https://user-images.githubusercontent.com/33888540/54401120-ae064600-469c-11e9-8795-f1efcadb6bd4.png)

**Comment by taylorjacklespriggs** [comment]
This issue persists on Firefox 68.0.1
https://codepen.io/plotly/pen/QpKOPr
<img width="687" alt="Screen Shot 2019-08-08 at 1 39 28 PM" src="https://user-images.githubusercontent.com/11512276/62736734-ee1aec00-b9e2-11e9-9d53-b4098a5353ac.png">

specs
MacOS Sierra 10.12.6
MacBook Pro (15-inch, 2017)
Graphics
 - Radeon Pro 555 2048 MB
 - Intel HD Graphics 630 1536 MB

**Comment by archmoj** [comment]
Fixed by #5500
